### PR TITLE
chore(turbo): delete Go related release code and config

### PR DESCRIPTION
**Pull request by mehulkar**
Closes TURBO-2214

### CODE REVIEW

**Review by sokra — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -846,7 +846,7 @@ jobs:
 
       - name: Lint
         # Filters some workspaces out:
-        # - `cli` because it runs golangci-lint and this Job doesn't have all the dependencies for that.
+        # - `cli` because (no reason anymore)
```
Comment (chris-olszewski) [inline .github/workflows/test.yml:849]: Should we stop filtering out `cli` then?

File: .github/workflows/turborepo-release.yml [hunk 2]
```diff
@@ -193,15 +193,6 @@ jobs:
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
 
-      - name: Install GoReleaser
```
Comment (chris-olszewski) [inline .github/workflows/turborepo-release.yml:196]: We still need this, we haven't ripped out `goreleaser` yet. `make publish-turbo` invokes it.

File: cli/Makefile [hunk 3]
```diff
@@ -118,9 +78,6 @@ publish-turbo: clean build
 
 	npm config set --location=project "//registry.npmjs.org/:_authToken" $(NPM_TOKEN)
 
-	# Publishes the native npm modules.
-	goreleaser release --rm-dist -f combined-shim.yml $(SKIP_PUBLISH)
-
```
Comment (chris-olszewski) [inline cli/Makefile:123]: I don't think we can get rid of this. It is executing some code that prepares the various platform packages.

**Review by chris-olszewski — CHANGES_REQUESTED** [review]
I'm almost positive we can't just remove our calls to `goreleaser` without doing some additional work ourselves

File: .github/workflows/test.yml (hunk 1 above)
Comment (mehulkar) [reply .github/workflows/test.yml:849]: separate PR, I don't know if linting is setup there at all

File: cli/Makefile (hunk 3 above)
Comment (mehulkar) [reply cli/Makefile]: brought back all the Makefile and `*.yml` and goreleaser stuff

File: .github/workflows/test.yml (hunk 1 above)
Comment (mehulkar) [reply .github/workflows/test.yml]: actually `lint` script is gone from there, so we can safely remove this `!cli`. Done

**Comment by chris-olszewski** [comment]
https://github.com/vercel/turbo/pull/7221 should fix the integration test failures

**Pull request merged**